drivers/ethernet/eth_xilinx_axienet.c

```diff
@@ -0,0 +1,577 @@
+
+#define LOG_MODULE_NAME eth_xilinx_axienet
+#define LOG_LEVEL       CONFIG_ETHERNET_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <sys/types.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/ethernet.h>
+#include <ethernet/eth_stats.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/barrier.h>
+
+#include "../dma/dma_xilinx_axi_dma.h"
+
+/* register offsets and masks */
+#define XILINX_AXIENET_INTERRUPT_STATUS_OFFSET         0x0000000C
+#define XILINX_AXIENET_INTERRUPT_STATUS_RXREJ_MASK     0x00000008
+#define XILINX_AXIENET_INTERRUPT_STATUS_RXFIFOOVR_MASK 0x00000010 /* Rx fifo overrun */
+#define XILINX_AXIENET_INTERRUPT_PENDING_OFFSET        0x00000010
+
+#define XILINX_AXIENET_INTERRUPT_PENDING_RXCMPIT_MASK     0x00000004 /* Rx complete */
+#define XILINX_AXIENET_INTERRUPT_PENDING_RXRJECT_MASK     0x00000008 /* Rx frame rejected */
+#define XILINX_AXIENET_INTERRUPT_PENDING_RXFIFOOVR_MASK   0x00000010 /* Rx fifo overrun */
+#define XILINX_AXIENET_INTERRUPT_PENDING_TXCMPIT_MASK     0x00000020 /* Tx complete */
+#define XILINX_AXIENET_INTERRUPT_PENDING_RXDCMLOCK_MASK   0x00000040 /* Rx Dcm Lock */
+#define XILINX_AXIENET_INTERRUPT_PENDING_MGTRDY_MASK      0x00000080 /* MGT clock Lock */
+#define XILINX_AXIENET_INTERRUPT_PENDING_PHYRSTCMPLT_MASK 0x00000100 /* Phy Reset complete */
+
+#define XILINX_AXIENET_INTERRUPT_PENDING_RXREJ_MASK 0x00000008
+#define XILINX_AXIENET_INTERRUPT_ENABLE_OFFSET      0x00000014
+#define XILINX_AXIENET_INTERRUPT_ENABLE_RXREJ_MASK  0x00000008
+#define XILINX_AXIENET_INTERRUPT_ENABLE_OVR_MASK    0x00000010 /* FIFO overrun */
+
+#define XILINX_AXIENET_RECEIVER_CONFIGURATION_WORD_0_REG_OFFSET     0x00000400
+#define XILINX_AXIENET_RECEIVER_CONFIGURATION_WORD_1_REG_OFFSET     0x00000404
+#define XILINX_AXIENET_RECEIVER_CONFIGURATION_WORD_1_REG_RX_EN_MASK 0x10000000
+#define XILINX_AXIENET_RECEIVER_CONFIGURATION_FLOW_CONTROL_OFFSET   0x0000040C
+#define XILINX_AXIENET_RECEIVER_CONFIGURATION_FLOW_CONTROL_EN_MASK  0x20000000
+#define XILINX_AXIENET_TX_CONTROL_REG_OFFSET                        0x00000408
+#define XILINX_AXIENET_TX_CONTROL_TX_EN_MASK                        (1 << 11)
+
+#define XILINX_AXIENET_UNICAST_ADDRESS_WORD_0_OFFSET 0x00000700
+#define XILINX_AXIENET_UNICAST_ADDRESS_WORD_1_OFFSET 0x00000704
+
+#define ETH_ALEN 6
```

Comment:
NET_ETH_ADDR_LEN macro already exists, please use that

Reply:
Done, must have overlooked this.